src/gui/socketapi.cpp

```diff
@@ -545,11 +545,37 @@ void SocketApi::command_GET_STRINGS(const QString &, SocketListener *listener)
     } };
     listener->sendMessage(QString("GET_STRINGS:BEGIN"));
     for (auto key_value : strings) {
-        listener->sendMessage(QString("STRING:%1:%2").arg(key_value.first, key_value.second));
+        if (argument.isEmpty() || argument == QLatin1String(key_value.first)) {
+            listener->sendMessage(QString("STRING:%1:%2").arg(key_value.first, key_value.second));
+        }
     }
     listener->sendMessage(QString("GET_STRINGS:END"));
 }
 
+void SocketApi::command_GET_MENU_ITEMS(const QString &argument, OCC::SocketListener *listener)
+{
+    listener->sendMessage(QString("GET_MENU_ITEMS:BEGIN"));
+    bool hasSeveralFiles = argument.contains(QLatin1Char('\x1e')); // Record Separator
```

Comment:
So while the extensions do support selecting and then querying multiple files, the client does not yet do I understand this correctly? @ogoffart @ckamm

Reply:
The placeholder branch does.

Reply:
Yep, you can ask to download multiple placeholders at once with this functionality :)